torch/nn/modules/rnn.py

```diff
@@ -171,6 +185,10 @@ def forward(self, input, hx=None):
                                  requires_grad=False)
             if self.mode == 'LSTM':
                 hx = (hx, hx)
+        else:
+            # Each batch of the hidden state should match the input sequence that
+            # the user believes he/she is passing in.
+            self.permute_hidden(hx, sorted_indices)
```

Comment:
I'm confused by this line. It looks like `permute_hidden` returns the result and does not modify its inputs, bu this line doesn't use the output.

Reply:
Good catch -- it should be `hx = self.permute_hidden`, I will update that

Reply:
Fixed and added a test for this case.